Sources/AppleTools/XcodebuildBasedTestRunner.swift

```diff
@@ -109,7 +111,12 @@ public final class XcodebuildBasedTestRunner: TestRunner {
     }
     
     public func additionalEnvironment(testRunnerWorkingDirectory: AbsolutePath) -> [String: String] {
-        return [XcodebuildTestRunnerConstants.envXcresultPath: testRunnerWorkingDirectory.appending(component: "resultBundle.xcresult").pathString]
-
+        return [
+            XcodebuildTestRunnerConstants.envXcresultPath: xcresultBundlePath(testRunnerWorkingDirectory: testRunnerWorkingDirectory).pathString
+        ]
+    }
+    
+    private func xcresultBundlePath(testRunnerWorkingDirectory: AbsolutePath) -> AbsolutePath {
+        return testRunnerWorkingDirectory.appending(component: "resultBundle.xcresult")
```

Comment:
This is a main change - this path now used from `additionalEnvironment` and `prepareToRun` methods. Previously the paths were different because of logic error.